src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -1110,35 +1110,43 @@ type Connector
         co.MaxConnectionLimit <- defaultArg gatewayModeMaxConnectionLimit 1000
         co
 
-    /// Yields an CosmosClient configured and Connect()ed to a given DocDB collection per the requested `discovery` strategy
-    let connect
-        (   /// Name should be sufficient to uniquely identify this connection within a single app instance's logs
-            name,
-            discovery : Discovery) : Async<Client.DocumentClient> =
-        let connect (uri: Uri, key: string) = async {
-            let name = String.concat ";" <| seq {
-                yield name
-                match tags with None -> () | Some tags -> for key, value in tags do yield sprintf "%s=%s" key value }
-            let sanitizedName = name.Replace('\'','_').Replace(':','_') // sic; Align with logging for ES Adapter
-            let client =
-                let consistencyLevel = Nullable(defaultArg defaultConsistencyLevel ConsistencyLevel.Session)
-                if defaultArg bypassCertificateValidation false then
-                    let inhibitCertCheck = new System.Net.Http.HttpClientHandler(ServerCertificateCustomValidationCallback = fun _ _ _ _ -> true)
-                    new Client.DocumentClient(uri, key, inhibitCertCheck, clientOptions, consistencyLevel) // overload introduced in 2.2.0 SDK
-                else new Client.DocumentClient(uri, key, clientOptions, consistencyLevel)
-            log.ForContext("Uri", uri).Information("CosmosDb Connection Name {connectionName}", sanitizedName)
-            do! client.OpenAsync() |> Async.AwaitTaskCorrect
-            return client }
-
-        match discovery with Discovery.UriAndKey(databaseUri=uri; key=key) -> connect (uri,key)
+    let logName (uri : Uri) name =
+        let name = String.concat ";" <| seq {
+            yield name
+            match tags with None -> () | Some tags -> for key, value in tags do yield sprintf "%s=%s" key value }
+        let sanitizedName = name.Replace('\'','_').Replace(':','_') // sic; Align with logging for ES Adapter
+        log.ForContext("Uri", uri).Information("CosmosDb Connection Name {connectionName}", sanitizedName)
+    let mkClient name skipLog = function
+        | Discovery.UriAndKey(databaseUri=uri; key=key) ->
+            if not skipLog then logName uri name
```

Comment:
🤔 the name should probably also be fed into the `DocumentClient` for inclusion in cosmosdb's analytics. I dont know how to do that - the `User-Agent` doesnt feel right for that

Reply:
someone'll bring it some day ;)